src/test/java/net/consensys/orion/http/handler/GetPrivacyGroupHandlerTest.java

```diff
@@ -0,0 +1,165 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class GetPrivacyGroupHandlerTest extends HandlerTest {
+  private MemoryKeyStore memoryKeyStore;
+  private String privacyGroupId;
+  private FakePeer fakePeer;
+  private Box.PublicKey senderKey;
+  private String[] members;
+
+  @Override
+  protected Enclave buildEnclave(final Path tempDir) {
+    memoryKeyStore = new MemoryKeyStore();
+    final Box.PublicKey defaultNodeKey = memoryKeyStore.generateKeyPair();
+    memoryKeyStore.addNodeKey(defaultNodeKey);
+    return new SodiumEnclave(memoryKeyStore);
+  }
+
+  @BeforeEach
+  void setup() throws IOException, InterruptedException {
+    senderKey = memoryKeyStore.generateKeyPair();
+    final Box.PublicKey recipientKey = memoryKeyStore.generateKeyPair();
+
+    members = new String[] {encodeBytes(senderKey.bytesArray()), encodeBytes(recipientKey.bytesArray())};
+    final PrivacyGroupRequest privacyGroupRequestExpected =
+        buildPrivacyGroupRequest(members, encodeBytes(senderKey.bytesArray()), "test", "desc");
+    final Request request = buildPrivateAPIRequest("/createPrivacyGroup", JSON, privacyGroupRequestExpected);
+
+    final byte[] privacyGroupPayload = enclave.generatePrivacyGroupId(
+        new Box.PublicKey[] {senderKey, recipientKey},
+        privacyGroupRequestExpected.getSeed().get(),
+        PrivacyGroupPayload.Type.PANTHEON);
+
+    // create fake peer
+    fakePeer = new FakePeer(new MockResponse().setBody(encodeBytes(privacyGroupPayload)), recipientKey);
+    networkNodes.addNode(Collections.singletonList(fakePeer.publicKey), fakePeer.getURL());
+
+    // execute request
+    final Response resp = httpClient.newCall(request).execute();
+
+    assertEquals(200, resp.code());
+
+    final RecordedRequest recordedRequest = fakePeer.server.takeRequest();
+    assertEquals("/pushPrivacyGroup", recordedRequest.getPath());
+    assertEquals("POST", recordedRequest.getMethod());
+
+    final PrivacyGroup privacyGroup = Serializer.deserialize(JSON, PrivacyGroup.class, resp.body().bytes());
+    privacyGroupId = privacyGroup.getPrivacyGroupId();
+  }
+
+  @Test
+  void retrievesPrivacyGroupDetails() throws Exception {
+    final GetPrivacyGroupRequest getPrivacyGroupRequest = buildGetPrivacyGroupRequest(privacyGroupId);
+    final Request request = buildPrivateAPIRequest("/getPrivacyGroup", JSON, getPrivacyGroupRequest);
+    fakePeer.addResponse(new MockResponse().setBody(privacyGroupId));
+
+    final Response resp = httpClient.newCall(request).execute();
+
+    assertThat(resp.code()).isEqualTo(200);
+    final PrivacyGroup privacyGroup = Serializer.deserialize(JSON, PrivacyGroup.class, resp.body().bytes());
+    assertThat(privacyGroup.getPrivacyGroupId()).isEqualTo(privacyGroupId);
+    assertThat(privacyGroup.getName()).isEqualTo("test");
+    assertThat(privacyGroup.getDescription()).isEqualTo("desc");
+    assertThat(privacyGroup.getMembers()).isEqualTo(members);
```

Comment:
these values all come from the setup - but its not super obvious that they should line up ... any idea how to make that more explicit?

Reply:
added some constants, hopefully that makes it more obvious